pandas/core/sparse/array.py

```diff
@@ -789,7 +790,15 @@ def make_sparse(arr, kind='block', fill_value=None):
         if is_string_dtype(arr):
             arr = arr.astype(object)
 
-        mask = arr != fill_value
+        if is_object_dtype(arr.dtype):
+            mask = np.ones(len(arr), dtype=np.bool)
```

Comment:
ok pls add a comment on what/why this is happening.

no general numpy routine to do this? (I don't think we have a pandas one), but have a look around.

Reply:
@jreback Added comments. I haven't found yet. I think there is no numpy general method.